<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1684</width>
    <height>965</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>171</width>
      <height>901</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <property name="spacing">
      <number>5</number>
     </property>
     <property name="sizeConstraint">
      <enum>QLayout::SetMaximumSize</enum>
     </property>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>端口号：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="PortBox"/>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPushButton" name="btnFindPort">
         <property name="text">
          <string>寻找端口</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>波特率：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="BaudBox">
           <item>
            <property name="text">
             <string>1200</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>2400</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>4800</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>9600</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>19200</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>38400</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>115200</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>数据位：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="BitNumBox">
           <item>
            <property name="text">
             <string>8</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>校验位：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="ParityBox">
           <item>
            <property name="text">
             <string>0</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>停止位：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="StopBox">
           <item>
            <property name="text">
             <string>1</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>2</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QPushButton" name="blueTooth">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>显示波形</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="openButton">
       <property name="text">
        <string>打开串口</string>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>接收窗口</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTextEdit" name="textEdit">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkBoxView">
         <property name="text">
          <string>接收显示</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>发送窗口</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTextEdit" name="textEdit_2">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QPushButton" name="clearButton">
           <property name="text">
            <string>清空接收</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="sendButton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>发送数据</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QComboBox" name="modelComboBox">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <item>
          <property name="text">
           <string>Steer</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Motor</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Ohter</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>580</y>
      <width>260</width>
      <height>331</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_8">
       <item>
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>Steer_P</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_10">
         <property name="text">
          <string>Steer_D</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>Diff</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_7">
       <item>
        <widget class="QDoubleSpinBox" name="steerPSpinBox">
         <property name="minimum">
          <double>0.000000000000000</double>
         </property>
         <property name="maximum">
          <double>10.000000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="steerDSpinBox">
         <property name="minimum">
          <double>0.000000000000000</double>
         </property>
         <property name="maximum">
          <double>100.000000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="diffSpinBox">
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="minimum">
          <double>0.000000000000000</double>
         </property>
         <property name="maximum">
          <double>0.020000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_10">
       <item>
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>Motor_P</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_12">
         <property name="text">
          <string>Motor_I</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_13">
         <property name="text">
          <string>Speed_Exp</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_11">
       <item>
        <widget class="QDoubleSpinBox" name="motorPSpinBox">
         <property name="maximum">
          <double>50.000000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="motorISpinBox">
         <property name="maximum">
          <double>50.000000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="speedSpinBox">
         <property name="maximum">
          <number>500</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QPushButton" name="PIDGet">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>获取</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="PIDSend">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>发送</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="startCarButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>StartCar</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>580</y>
      <width>95</width>
      <height>331</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_5">
     <item>
      <widget class="QPushButton" name="variableButton">
       <property name="text">
        <string>变量</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="sdButton">
       <property name="text">
        <string>SD</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGraphicsView" name="ccd1GraphicsView">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>600</y>
      <width>491</width>
      <height>31</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
   </widget>
   <widget class="QGraphicsView" name="ccd2GraphicsView">
    <property name="geometry">
     <rect>
      <x>1160</x>
      <y>600</y>
      <width>491</width>
      <height>31</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
   </widget>
   <widget class="QCheckBox" name="ccd1CheckBox">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>580</y>
      <width>91</width>
      <height>19</height>
     </rect>
    </property>
    <property name="text">
     <string>ccd1</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="ccd2CheckBox">
    <property name="geometry">
     <rect>
      <x>1160</x>
      <y>580</y>
      <width>91</width>
      <height>19</height>
     </rect>
    </property>
    <property name="text">
     <string>ccd2</string>
    </property>
   </widget>
   <widget class="QChartView" name="ccd1GrayView">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>640</y>
      <width>551</width>
      <height>271</height>
     </rect>
    </property>
   </widget>
   <widget class="QChartView" name="ccd2GrayView">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>640</y>
      <width>551</width>
      <height>271</height>
     </rect>
    </property>
   </widget>
   <widget class="QChartView" name="dataView">
    <property name="geometry">
     <rect>
      <x>185</x>
      <y>11</y>
      <width>951</width>
      <height>561</height>
     </rect>
    </property>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>1149</x>
      <y>9</y>
      <width>531</width>
      <height>551</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="4">
      <widget class="QLabel" name="label_16">
       <property name="text">
        <string>小S速度</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLabel" name="label_17">
       <property name="text">
        <string>进弯速度</string>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QSpinBox" name="CurveSpeed"/>
     </item>
     <item row="1" column="3">
      <widget class="QSpinBox" name="outCurveSpeed"/>
     </item>
     <item row="1" column="0">
      <widget class="QSpinBox" name="straightSpeed"/>
     </item>
     <item row="3" column="3">
      <widget class="QLabel" name="label_23">
       <property name="text">
        <string>进弯时间</string>
       </property>
      </widget>
     </item>
     <item row="3" column="4">
      <widget class="QLabel" name="label_24">
       <property name="text">
        <string>小弯速度</string>
       </property>
      </widget>
     </item>
     <item row="4" column="3">
      <widget class="QSpinBox" name="intoCurveTime"/>
     </item>
     <item row="7" column="2">
      <widget class="QSpinBox" name="CurveThreshold"/>
     </item>
     <item row="4" column="4">
      <widget class="QSpinBox" name="litteCurveSpeed"/>
     </item>
     <item row="6" column="1">
      <widget class="QLabel" name="label_25">
       <property name="text">
        <string>进弯阈值</string>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <widget class="QLabel" name="label_19">
       <property name="text">
        <string>直道阈值</string>
       </property>
      </widget>
     </item>
     <item row="6" column="2">
      <widget class="QLabel" name="label_26">
       <property name="text">
        <string>弯道阈值</string>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QSpinBox" name="obstacleSpeed"/>
     </item>
     <item row="4" column="1">
      <widget class="QSpinBox" name="rampUpSpeed"/>
     </item>
     <item row="5" column="0" colspan="5">
      <widget class="QPushButton" name="speedSend">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>发送速度</string>
       </property>
      </widget>
     </item>
     <item row="9" column="0">
      <widget class="QSpinBox" name="obstacleThreshold"/>
     </item>
     <item row="9" column="1">
      <widget class="QSpinBox" name="rampThreshold"/>
     </item>
     <item row="9" column="2">
      <widget class="QSpinBox" name="rampDownThreshold"/>
     </item>
     <item row="9" column="3">
      <widget class="QSpinBox" name="outCurveThreshold_2"/>
     </item>
     <item row="9" column="4">
      <widget class="QSpinBox" name="littleCurveThreshold"/>
     </item>
     <item row="0" column="2">
      <widget class="QLabel" name="label_14">
       <property name="text">
        <string>弯道速度</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLabel" name="label_21">
       <property name="text">
        <string>上坡速度</string>
       </property>
      </widget>
     </item>
     <item row="11" column="0" colspan="5">
      <widget class="QPushButton" name="sendThreshold">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>发送阈值</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="label_15">
       <property name="text">
        <string>直道速度</string>
       </property>
      </widget>
     </item>
     <item row="4" column="2">
      <widget class="QSpinBox" name="rampDownSpeed"/>
     </item>
     <item row="6" column="3">
      <widget class="QLabel" name="label_27">
       <property name="text">
        <string>出弯阈值</string>
       </property>
      </widget>
     </item>
     <item row="8" column="4">
      <widget class="QLabel" name="label_33">
       <property name="text">
        <string>小弯阈值</string>
       </property>
      </widget>
     </item>
     <item row="12" column="0" colspan="5">
      <widget class="QPushButton" name="allSend">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>全部发送</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_20">
       <property name="text">
        <string>障碍速度</string>
       </property>
      </widget>
     </item>
     <item row="0" column="3">
      <widget class="QLabel" name="label_18">
       <property name="text">
        <string>出弯速度</string>
       </property>
      </widget>
     </item>
     <item row="7" column="0">
      <widget class="QSpinBox" name="straightThreshold"/>
     </item>
     <item row="6" column="4">
      <widget class="QLabel" name="label_28">
       <property name="text">
        <string>小S阈值</string>
       </property>
      </widget>
     </item>
     <item row="7" column="1">
      <widget class="QSpinBox" name="intoCurveThreshold"/>
     </item>
     <item row="7" column="3">
      <widget class="QSpinBox" name="outCurveThreshold"/>
     </item>
     <item row="7" column="4">
      <widget class="QSpinBox" name="snakeThreshold"/>
     </item>
     <item row="1" column="4">
      <widget class="QSpinBox" name="snakeSpeed"/>
     </item>
     <item row="8" column="0">
      <widget class="QLabel" name="label_29">
       <property name="text">
        <string>障碍阈值</string>
       </property>
      </widget>
     </item>
     <item row="8" column="1">
      <widget class="QLabel" name="label_30">
       <property name="text">
        <string>上坡阈值</string>
       </property>
      </widget>
     </item>
     <item row="8" column="2">
      <widget class="QLabel" name="label_31">
       <property name="text">
        <string>下坡阈值</string>
       </property>
      </widget>
     </item>
     <item row="8" column="3">
      <widget class="QLabel" name="label_32">
       <property name="text">
        <string>出弯阈值</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QSpinBox" name="intoCurveSpeed"/>
     </item>
     <item row="3" column="2">
      <widget class="QLabel" name="label_22">
       <property name="text">
        <string>下坡速度</string>
       </property>
      </widget>
     </item>
     <item row="10" column="0" colspan="5">
      <widget class="QPushButton" name="getInfomation">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>获取信息</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1684</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QChartView</class>
   <extends>QGraphicsView</extends>
   <header location="global">QtCharts/QChartView</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
